source/common/stats/tag_utility.cc

```diff
@@ -48,6 +51,29 @@ SymbolTable::StoragePtr TagStatNameJoiner::joinNameAndTags(StatName name,
 
   return symbol_table.join(stat_names);
 }
+
+bool isTagValueValid(absl::string_view name) {
+  std::regex regex{Config::NAME_REGEX, std::regex::optimize};
```

Comment:
In the existing code, NAME_REGEX is evaluated by RE2, not std::regex, so if you want consistency you should. use RE2 to test it.